js-api-spec/value/color.test.ts

```diff
@@ -431,3 +1674,536 @@ describe('SassColor', () => {
+
+    describe(space.name, () => {
+      let color: SassColor, blue: SassColor;
+      beforeEach(() => {
+        color = space.constructor(...space.pink);
+        blue = space.constructor(...space.blue);
+      });
+      it('is a value', () => {
+        expect(color).toBeInstanceOf(Value);
+      });
+
+      it('is a color', () => {
+        expect(color).toBeInstanceOf(SassColor);
+        expect(color.assertColor()).toBe(color);
+      });
+
+      it("isn't any other type", () => {
+        expect(() => color.assertBoolean()).toThrow();
+        expect(() => color.assertCalculation()).toThrow();
+        expect(() => color.assertFunction()).toThrow();
+        expect(() => color.assertMap()).toThrow();
+        expect(color.tryMap()).toBe(null);
+        expect(() => color.assertMixin()).toThrow();
+        expect(() => color.assertNumber()).toThrow();
+        expect(() => color.assertString()).toThrow();
+      });
+
+      it('throws on invalid alpha', () => {
+        expect(() => {
+          space.constructor(...space.pink, -1);
+        }).toThrow();
+        expect(() => {
+          space.constructor(...space.pink, 1.1);
+        }).toThrow();
+      });
+
+      if (space.channels.includes('lightness')) {
+        describe('out of range lightness', () => {
+          it('throws on negative lightness', () => {
+            const index = space.channels.findIndex(
+              channel => channel === 'lightness'
+            );
+            const channels = [...space.pink] as [number, number, number];
+            channels[index] = -1;
+            expect(() => space.constructor(...channels)).toThrow();
+          });
+          // TODO: Failing for oklab and oklch in dart-sass
+          it('throws on lightness higher than bounds', () => {
+            const index = space.channels.findIndex(
+              channel => channel === 'lightness'
+            );
+            const channels = [...space.pink] as [number, number, number];
+            channels[index] = space.ranges[index][1] + 1;
+            expect(() => space.constructor(...channels)).toThrow();
+          });
+        });
+      }
+
+      it(`returns name for ${space.name}`, () => {
+        expect(color.space).toBe(space.name);
+      });
+      // Space conversions in ColorJS are close but not precise enough to match
+      skipForImpl('sass-embedded', () => {
+        describe('toSpace', () => {
+          spaceNames.forEach(destinationSpaceId => {
+            it(`converts pink to ${destinationSpaceId}`, () => {
+              const destinationSpace = spaces[destinationSpaceId];
+              const res = color.toSpace(destinationSpace.name);
+              expect(res.space).toBe(destinationSpace.name);
+
+              const expected = destinationSpace.constructor(
+                ...destinationSpace.pink
+              );
+              expect(res).toEqualWithHash(expected);
+            });
+          });
+        });
+      });
```

Comment:
@nex3 Using ColorJS to do color conversions works, but the math isn't precise enough to pass these tests, e.g.:

```
expected color(xyz-d50 0.66406985168935 0.5367266828656 0.43459579340121) to be .equal() to color(xyz-d50 0.6640698533004 0.53672666252811 0.43459582467203)
```

I'm not sure if we want to relax the precision requirements, or have different precision requirements for different platforms, or...?

Reply:
Resolved in https://github.com/sass/sass-spec/pull/1949/commits/d5d18b6a3d704f1277ebcbff14161dc22456f82c